### PR TITLE
Correctly marks all required input fields on incidents page

### PR DESCRIPTION
**Pull request by podliashanyk**
Changes made:

-  Filter title is required.
-  Acknowledgment and closing messages are no longer required.
-  Ticket URL if typed, must be a valid absolute URL.
-  NB! Ticket URL if left blank, removes a ticket from the incident (this functionality was there beforehand, it is now enhanced and made clear to the user).
- Update labels and titles on incidents select to reflect the actual action.

UI changes that reflect restrictions:

- All required fields are marked with asterisk.
- If a provided filter name is a blank string, the input field is highlighted in red and helper text appears. No request is made to the server unless a valid name is provided.
- If a provided URL is not a valid absolute URL,  the input field is highlighted in red and helper text appears. No request is made to the server unless a valid absolute URL is provided.

### CODE REVIEW

File: src/components/incident/AddTicketUrlSignOffAction.tsx [hunk 1]
```diff
@@ -23,6 +23,10 @@ const AddTicketUrl: React.FC<AddTicketUrlPropsType> = ({
       dialogSubmitText="Submit"
       dialogCancelText="Cancel"
       dialogButtonText="Add ticket"
+      dialogInputLabel="Valid ticket URL"
+      isDialogInputRequired={false}
+      dialogInputType="url"
+      dialogHelperText="By leaving this field empty, you remove any existing tickets from the selected incidents"
```
Comment (hmpf) [inline src/components/incident/AddTicketUrlSignOffAction.tsx:29]: Clunky text but I can't think of anything better right now. The tickets still exist, it's just the link that is removed.

"Leave this field empty in order to remove ticket urls from the selected incidents" maybe? Hm.

File: src/components/incident/IncidentDetails.tsx [hunk 2]
```diff
@@ -98,14 +98,10 @@ type TagChipPropsType = {
...
-  // https://stackoverflow.com/questions/5717093/check-if-a-javascript-string-is-a-url
-  try {
-    new URL(url);
-  } catch (_) {
-    return false;
-  }
-  return true;
+  let pattern = new RegExp(
+    "^(http(s)?:\\/\\/)+[\\w\\-\\._~:\\/?#[\\]@!\\$&'\\(\\)\\*\\+,;=.]+$",
+    'i');
```
Comment (hmpf) [inline src/components/incident/IncidentDetails.tsx:103]: Bare regexps are a code smell. If you got this one from somewhere else, citation needed. If not, you need a *large* test-suite documenting the urls this regexp was designed for: With path, with query, with php-style queries etc.

**Review by hmpf — CHANGES_REQUESTED** [review]
Some nitpicks, I also want @lunkwill42 to have a look at this one.

File: src/components/incident/AddTicketUrlSignOffAction.tsx (hunk 1 above)
Comment (podliashanyk) [reply src/components/incident/AddTicketUrlSignOffAction.tsx:29]: I fix!

File: src/components/incident/IncidentDetails.tsx [hunk 3]
```diff
@@ -98,10 +99,14 @@ type TagChipPropsType = {
...
-    'i');
-  return Boolean(pattern.test(url));
+  // Pavlo's answer at
+  // https://stackoverflow.com/questions/5717093/check-if-a-javascript-string-is-a-url
+  try {
+    new URL(url);
+  } catch (_) {
+    return false;
+  }
+  return true;
```
Comment (hmpf) [inline src/components/incident/IncidentDetails.tsx]: This duplicates `validateUrlInput` in 'src/utils.tsx'. Pick one.

File: src/utils.tsx [hunk 4]
```diff
@@ -263,8 +263,12 @@ export function validateStringInput(value: string): Boolean {
...
-    'i');
-  return Boolean(pattern.test(value));
+  // Pavlo's answer at
+  // https://stackoverflow.com/questions/5717093/check-if-a-javascript-string-is-a-url
+  try {
+    new URL(value);
+  } catch (_) {
+    return false;
+  }
+  return true;
```
Comment (hmpf) [inline src/utils.tsx:273]: This duplicates `isValidUrl` in `src/components/incident/IncidentDeatils.tsx`. Pick one. I guess it is better to import from here into the latter, but `isValidUrl` is a better name than `validateUrlInput`. You could also have `validateUrlInput` reuse `isValidUrl`.

File: src/components/incident/SignOffAction.tsx [hunk 5]
```diff
@@ -116,11 +116,6 @@ const SignOffAction: React.FC<SignOffActionPropsType> = ({
             id="message"
             label={dialogInputLabel}
             type={dialogInputType}
-            inputProps={
-              dialogInputType === "url" ?
-                { pattern: "^(http(s)?:\\/\\/)+[\\w\\-\\._~:\\/?#[\\]@!\\$&'\\(\\)\\*\\+,;=.]+$" } :
-                {}
-            }
```
Comment (hmpf) [inline src/components/incident/SignOffAction.tsx]: Good, duplicate regular expressions sprinkled through the code is a BIG code smell.

**Review by hmpf — CHANGES_REQUESTED** [review]

**Comment by podliashanyk** [comment]
Out of two identical methods kept the one in utils.tsx, but renamed it to isValidUrl.

**Comment by lunkwill42** [comment]
Just to clarify, I accidentally pushed a commit to this branch by mistake and rolled it back to 3f5ebf4

**Review by lunkwill42 — CHANGES_REQUESTED** [review]
Nice work :)

One issue remains, that I can spot:

- [ ] Creating a new timeslot appears to require a name (has an asterisk), but the UI still allows saving with an empty name, causing a 400 error from the backend.

**Comment by podliashanyk** [comment]
> One issue remains, that I can spot:
> 
> * [ ]  Creating a new timeslot appears to require a name (has an asterisk), but the UI still allows saving with an empty name, causing a 400 error from the backend.

This PR fixes required fields on incidents page only. Required fileds on the Timeslots page er still a to-do. 🙂 
Or do you mean that I should fix it in this PR?

**Comment by hmpf** [comment]
> > Creating a new timeslot appears to require a name (has an asterisk), but the UI still allows saving with an empty name, causing a 400 error from the backend.
> 
> This PR fixes required fields on incidents page only. Required fileds on the Timeslots page are still a to-do. slightly_smiling_face
> Or do you mean that I should fix it in this PR?

Naw, let's merge this now, but make a new issue for the timeslots-page and put it in the kanban-board so we don't forget.

**Comment by podliashanyk** [comment]
> > > Creating a new timeslot appears to require a name (has an asterisk), but the UI still allows saving with an empty name, causing a 400 error from the backend.
> > 
> > 
> > This PR fixes required fields on incidents page only. Required fileds on the Timeslots page are still a to-do. slightly_smiling_face
> > Or do you mean that I should fix it in this PR?
> 
> Naw, let's merge this now, but make a new issue for the timeslots-page and put it in the kanban-board so we don't forget.

Done! #287

**Review by hmpf — APPROVED** [review]

**Comment by lunkwill42** [comment]
> Naw, let's merge this now, but make a new issue for the timeslots-page and put it in the kanban-board so we don't forget.

Agreed :)

**Review by lunkwill42 — APPROVED** [review]

**Pull request merged**